.gitpod.Dockerfile

```diff
@@ -0,0 +1,114 @@
+    NODE_DEPS_PATH="${GITPOD_HOME}/node_modules"
+
+# -----------------------------------------------------------------------------
+# ---- Creating as root - note: make sure to change to gitpod in the end ----
+USER root
+
+# hadolint ignore=DL3008
+RUN apt-get update && \
+    apt-get install -yq --no-install-recommends \
+    # software-properties-common \
+    # locales \
+    # wget \
+    build-essential \
+    ca-certificates \
+    curl \
+    dirmngr \
+    make && \
+    # this needs to be done after installing dirmngr
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-key C99B11DEB97541F0 && \
+    apt-add-repository https://cli.github.com/packages && \
+    apt-get install -yq --no-install-recommends \
+    gh && \
+    locale-gen en_US.UTF-8 && \
+    # clean apt cache
+    apt-get clean && \
+    rm -rf /var/cache/apt/* &&\
+    rm -rf /var/lib/apt/lists/* &&\
+    rm -rf /tmp/*
+
+# Allows this Dockerfile to activate conda environments
+SHELL ["/bin/bash", "--login", "-o", "pipefail", "-c"]
+
+# -----------------------------------------------------------------------------
+# ---- Installing mamba  ----
+RUN wget -q -O mambaforge3.sh \
+    "https://github.com/conda-forge/miniforge/releases/download/$MAMBAFORGE_VERSION/Mambaforge-$MAMBAFORGE_VERSION-Linux-x86_64.sh" && \
+    bash mambaforge3.sh -p ${CONDA_DIR} -b && \
+    rm mambaforge3.sh && \
+    chown -R gitpod:gitpod ${CONDA_DIR}
+
+
+# -----------------------------------------------------------------------------
+# ---- Copy conda and config files ----
+# Copy conda environment file into the container - this needs to exists inside
+# the container to create a conda environment from it
+# basic workspace configurations
+COPY ./tools/gitpod/workspace_config /usr/local/bin/workspace_config
+RUN chmod a+rx /usr/local/bin/workspace_config && \
+    workspace_config
+
+# COPY ./tools/environment.yml /tmp/environment.yml
+
+# -----------------------------------------------------------------------------
+# ---- Create conda environment with base dependencies ----
+# Install dependencies
+RUN mamba create -n a11y-tests -c conda-forge python=3.9 nox pyyaml 'nodejs>=16,<17' yarn && \
+    conda activate ${CONDA_ENV}  && \
+    conda clean --all -f -y && \
+    rm -rf /tmp/* && \
+    echo node --version ">" `node --version`
+
+# -----------------------------------------------------------------------------
+# ---- Copy needed files for npm dependencies ----
+# COPY ./tests/retrolab/package.json package.json
+
+# Only install chromium for now - can change at build time
+# ARG BROWSERS="chromium"
+
+# RUN mkdir ${PLAYWRIGHT_BROWSERS_PATH}
+
+# RUN conda activate ${CONDA_ENV} && \
+#     rm -rf ${NODE_DEPS_PATH} && \
+#     npm install -g node-gyp && \
+#     chmod -R 777 ${PLAYWRIGHT_BROWSERS_PATH} && \
+#     npx playwright install-deps && \
+#     npx playwright install ${BROWSERS} && \
+#     # rm package.json  && \
+#     rm -rf /var/lib/apt/lists/*
+
```

Comment:
Some more thoughts on this.

It's unclear to me if we can install Playwright in a single place, using a directory structure like the following:

```
tests/
    package.json
    playwright.config.ts
    jupyterlab/
        test/
            a11y.test.ts
    retrolab/
        test./
            a11y.test.ts
```

or if each project that we want to test (JupyterLab app, JupyterLab docs, RetroLab app, etc.), will need to have its own Playwright config:

```
tests/
    jupyterlab/
        package.json
        playwright.config.ts
        test/
            a11y.test.ts
    retrolab/
        package.json
        playwright.config.ts
        test./
            a11y.test.ts
```

Reply:
> I commented this code out because I'm not sure we can have a single environment for the tests.
> 
> I mean, for example, I've never tried installing JupyterLab and RetroLab in the same environment, but I guess it wouldn't surprise me if that causes issues. Or if we think about testing the docs, we might use Pa11y instead of Axe+Playwright.

There are other ways to deal with this problem - added more thoughts in <https://github.com/jupyter/accessibility/pull/83#issuecomment-1123453188>
but I would argue that any "core" dependencies that will be reused across envs/tests/workflows should be part of the base Docker image

> So I took this part out of the Dockerfile and let it live in the noxfile instead.
Are we still using nox? - think we need to discuss this in depth again since we last discussed this was way too much overhead